### PR TITLE
Add a "Check that everything is fine" page

### PR DESCRIPTION
**Pull request by fxthomas**
This is a first attempt with some of the ideas in #1082, implemented just like the "About" page.

It's not finished yet, but it should be functional enough for testing so I'll leave it here for comments.

Screenshot:

Screen Shot 2020-01-05 at 18 06 08-fullpage

### CODE REVIEW

File: airsonic-main/src/main/java/org/airsonic/player/controller/InternalHelpController.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        map.put("statLastScanDate", indexManager.getStatistics().getScanDate());
+        map.put("statTotalDurationSeconds", indexManager.getStatistics().getTotalDurationInSeconds());
+        map.put("statTotalLengthBytes", FileUtils.byteCountToDisplaySize(indexManager.getStatistics().getTotalLengthInBytes()));
+
+        // Lucene index statistics
+        try (IndexReader reader = indexManager.getSearcher(IndexType.SONG).getIndexReader()) {
+            map.put("indexSongCount", reader.numDocs());
+            map.put("indexSongDeletedCount", reader.numDeletedDocs());
+        } catch (IOException e) {
+            LOG.debug("Unable to gather information", e);
```
Comment (jvoisin) [inline airsonic-main/src/main/java/org/airsonic/player/controller/InternalHelpController.java]: This should be changed for less generic log message :)

**Comment by jvoisin** [comment]
This looks amazing!

**Comment by fxthomas** [comment]
Thanks :wink: I added info about ffmpeg + ffprobe in addition to a small refactoring.

Screenshot from 2020-01-07 22-49-45

Now this just needs tweaking log messages, adding the correct i18n and testing on other databases (not sure if what I did works on PostgreSQL/MySQL).

**Comment by fxthomas** [comment]
I moved a few things around, tested on PostgreSQL, added i18n and a few additional consistency checks.

Should be good for review and testing now!

**Comment by jvoisin** [comment]
This looks glorious, and it would be a shame to let it rot: can we merge it?

**Comment by fxthomas** [comment]
Thanks!

I did some more tests on this to make sure it's going to be robust on recent PostgreSQL/MariaDB/MySQL versions, hope you don't mind the last minute bits:

* 27a3a7e: Don't check db size on filesystem in non-`legacy` profiles (we don't know how to do that in a consistent manner)
* 8734aa6: Fix a crash with MariaDB (which I hadn't tested earlier)
* 3e12c71: Add the server version in the page
* 6c8d133: Add simple tests for admin users and non-admin users

All previous commits are the same as what you reviewed, squashed and rebased on master.

**Comment by fxthomas** [comment]
With that it should be fully tested on all 3 major databases and ready to merge whenever you're okay with it.

**Comment by fxthomas** [comment]
Added one more last minute fix for the last failure. Obviously I should know that saying "everything is good" before Travis runs is a huge death flag... :rofl: 

Not sure why it's complaining about `cxf-gen` though?

**Comment by fxthomas** [comment]
I cleared Travis's cache, hopefully this fixes the build?

**Pull request merged**